plugins/modules/ipam_subnet.py

```diff
@@ -0,0 +1,2848 @@
+                    type: str
+                    returned: Always
+                utilization:
+                    description:
+                        - "The percentage of used IP addresses relative to the total IP addresses available in the scope of the object."
+                    type: int
+                    returned: Always
+        utilization_v6:
+            description:
+                - "The utilization of IPV6 addresses in the subnet."
+            type: dict
+            returned: Always
+            contains:
+                abandoned:
+                    description: ""
+                    type: str
+                    returned: Always
+                dynamic:
+                    description: ""
+                    type: str
+                    returned: Always
+                static:
+                    description: ""
+                    type: str
+                    returned: Always
+                total:
+                    description: ""
+                    type: str
+                    returned: Always
+                used:
+                    description: ""
+                    type: str
+                    returned: Always
+"""  # noqa: E501
+
+from ansible_collections.infoblox.bloxone.plugins.module_utils.modules import BloxoneAnsibleModule
+
+try:
+    from bloxone_client import ApiException, NotFoundException
+    from ipam import Subnet, SubnetApi
+except ImportError:
+    pass  # Handled by BloxoneAnsibleModule
+
+
+class SubnetModule(BloxoneAnsibleModule):
+    def __init__(self, *args, **kwargs):
+        super(SubnetModule, self).__init__(*args, **kwargs)
+
+        exclude = ["state", "csp_url", "api_key", "id"]
+        self._payload_params = {k: v for k, v in self.params.items() if v is not None and k not in exclude}
+        self._payload = Subnet.from_dict(self._payload_params)
+        self._existing = None
+
+    @property
+    def existing(self):
+        return self._existing
+
+    @existing.setter
+    def existing(self, value):
+        self._existing = value
+
+    @property
+    def payload_params(self):
+        return self._payload_params
+
+    @property
+    def payload(self):
+        return self._payload
+
+    def payload_changed(self):
+        if self.existing is None:
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = SubnetApi(self.client).read(self.params["id"], inherit="full")
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "absent":
+                    return None
+                raise e
+        else:
+            filter = f"address=='{self.params['address']}' and space=='{self.params['space']}'"
```

Comment:
this should also include cidr as well i think. Right now, this will only work if the address has the cidr in it, but the cidr can be in separate field as well.